UOP1_Project/Assets/Scripts/Sound/SoundManager.cs

```diff
@@ -6,43 +6,51 @@
 
 public class SoundManager : MonoBehaviour
 {
-	public static SoundManager singleton;
+	public static SoundManager Instance;
 
 	[Header ("Init Variables")]
 	[Tooltip ("Amount of sound emitters created on Start")]
-	public int initialPoolSize = 1;
-	[Tooltip ("Frequency at which the sound manager will delete extra emitters that are not in use")]
-	public float trimFrequency = 5;
-	[Tooltip ("Extra time after instanced emitters are not in use anymore before destroying them")]
-	public float extraTimeBeforeTrim = 5;
+	[SerializeField]
+	private int initialPoolSize = 1;
+	[SerializeField]
+	private SoundEmitter soundEmitterPrefab = default;
 
-	[Header ("Mixer Groups")]
-	public AudioMixerGroup masterMixer;
-	public AudioMixerGroup sfxMixer;
-	public AudioMixerGroup musicMixer;
+	SoundEmitterFactorySO _factory;
+	SoundEmitterPoolSO _pool;
+	public SoundEmitterPoolSO Pool { get { return _pool; } }
 
-	private List<SoundEmitter> _soundEmitterPool = new List<SoundEmitter>();
-	private GameObject _soundEmittersContainer;
+	[RuntimeInitializeOnLoadMethod(RuntimeInitializeLoadType.SubsystemRegistration)]
+	static void Init()
+	{
+		Instance = default;
+	}
 
 	private void Awake()
 	{
-		DontDestroyOnLoad(this);
-
-		if (!SoundManager.singleton)
-			singleton = this;
-		else
+		if (SoundManager.Instance)
 		{
-			Debug.LogWarning("One sound manager already present in the game, destroying " + this.name + ".");
+			Debug.LogWarning($"One sound manager already present in the game, destroying {name}.");
 			Destroy(this);
 		}
-	}
+		else
+		{
+			Instance = this;
+			DontDestroyOnLoad(this);
 
-	void Start()
-    {
-		InitEmitterPool();
+			InitPool();
+		}
+	}
 
-		InvokeRepeating("TrimUnusedEmitters", trimFrequency, trimFrequency);
-    }
+	private void InitPool()
+	{
+		_factory = ScriptableObject.CreateInstance<SoundEmitterFactorySO>();
+		_factory.Prefab = soundEmitterPrefab;
+		_factory.Prefab.name = "SoundEmitter Factory";
+		_pool = ScriptableObject.CreateInstance<SoundEmitterPoolSO>();
+		_pool.name = "SoundEmitter Pool";
+		_pool.Factory = _factory;
+		_pool.InitialPoolSize = initialPoolSize;
```

Comment:
I realize this is the example code given in the PR and LocalPoolTester example, but just so you know, you can create these factories and pools as assets of the project and plug them into the inspector as object references. Unless you _need_ to create these at runtime, I'd definitely recommend creating as project assets.